### PR TITLE
Supporting K2 with Skate + update to IJ 2024.2.1

### LINKED ISSUE #735: Support K2 in Skate
image

### PR DESCRIPTION
**Pull request by kateliu20**
Resolves #735

### CODE REVIEW

**Review by ZacSweers — APPROVED** [review]

**Comment by ZacSweers** [comment]
Uh oh, I think we have an infinite loop in a test

**Comment by kateliu20** [comment]
> Uh oh, I think we have an infinite loop in a test

Failed after 6 hours 🥲 Could it have something to do with this? Should we update the source compatibility to 21?

- The Java configuration specifies sourceCompatibility='17' but IntelliJ Platform '2024.2.1' requires sourceCompatibility='21'.

**Comment by kateliu20** [comment]
Or actually, when I run `./gradlew check` locally it's getting stuck on `SkateTraceReporterTest`. Maybe this is the test that's in the infinite loop?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -40,7 +40,7 @@ jobs:
 
       - name: Build and run tests
         id: gradle
-        run: ./gradlew check
+        run: ./gradlew check testIdeUi
```
Comment (ZacSweers) [inline .github/workflows/ci.yml:43]:
```suggestion
        run: ./gradlew check
```

**Pull request merged**